cpg_workflows/targets.py

```diff
@@ -97,18 +97,118 @@ def rich_id_map(self) -> dict[str, str]:
         return {s.id: s.rich_id for s in self.get_sequencing_groups() if s.participant_id != s.id}
 
 
+class MultiCohort(Target):
+    """
+    Represents a "multi-cohort" target - multiple cohorts in the workflow.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+
+        # NOTE: For a cohort, we simply pull the dataset name from the config.
+        input_cohorts = get_config()['workflow'].get('input_cohorts', [])
+        if input_cohorts:
+            cohorts_string = '_'.join(sorted(input_cohorts))
+
+        self.name = cohorts_string or get_config()['workflow']['dataset']
+
+        assert self.name, 'Ensure cohorts or dataset is defined in the config file.'
+
+        self._cohorts_by_name: dict[str, Cohort] = {}
+        self.analysis_dataset = Dataset(name=get_config()['workflow']['dataset'])
+
+    def __repr__(self):
+        return f'MultiCohort({len(self.get_cohorts())} cohorts)'
+
+    @property
+    def target_id(self) -> str:
+        """Unique target ID"""
+        return self.name
+
+    def get_cohorts(self, only_active: bool = True) -> list['Cohort']:
+        """
+        Gets list of all cohorts.
+        Include only "active" cohorts (unless only_active is False)
+        """
+        cohorts = [cohort for _name, cohort in self._cohorts_by_name.items()]
```

Comment:
Could use `.values()` here if you don't need `_name`, but no biggie.

(Or even just `list(self._cohorts_by_name.values())` or leave it as the view and make it into a list with `return list(cohorts)` only at the end.)

Reply:
Yeah totally fair enough, updated to use .values()! Briefly experimented with a one line comprehension for this entire method, but the readability was gross hahaha